collaborative-filter.Rmd

```diff
@@ -48,14 +56,36 @@ This is a basic collaborative filter! You have used information about interest a
 
 Now create a unit-based, rather than student-based similarity matrix for difficulty. Then use your similarity matrix to provide a suggested next unit to a student who is looking for the unit that is most similar in terms of difficulty to the "prediction" unit. 
 ```{r}
-
+d.SIM <- cosine(d)
+diag(d.SIM) <- NA
+head(rownames(d.SIM[order(d.SIM["pred.dif",], decreasing = TRUE),]), n = 1) 
 ```
 
 Finally, educational settings have important differences to purely commercial settings such as film or product suggestions. In education we want people not to just follow their interests as they may simply choose things that are easy for them so they learn very little. To reduce this possibility with your collaborative filter create a composite measure from interest and difficulty, then construct a similarity matrix using this measure. (HINT: PCA). Once you have built the similarity matrix generate a suggestion for a student who has just completed the "prediction" unit.
 
 ```{r}
-
-
+library(tidyr)
+i1 <- gather(interest,unit,interest, 2:7)
+d1 <- gather(diff, name, difficulty)
+c1 <- data.frame(i1$name, i1$unit, i1$interest, d1$difficulty)
+names(c1) <- c("name", "unit", "interest", "difficulty")
+c1 <- filter(c1, difficulty > 0)
+c2 <- select(c1, "interest", "difficulty")
+pc <- prcomp(c2)
+c3 <- data.frame(c1$name, c1$unit, pc$x)
+c4 <- select(c3, c1.name, c1.unit, PC1)
+c4$c1.unit <- gsub(".int", "", c4$c1.unit)
+c5 <- spread(c4, c1.name, PC1)
+row.names(c5) <- c5$c1.unit
+c5$c1.unit <- NULL
+c5 <- as.matrix(c5)
+c5 <- ifelse(is.na(c5), 0, c5)
+c5 <- t(c5)
+#Generating cosine similarity matrix for the units
+c.SIM <- cosine(c5)
+diag(c.SIM) <- NA
+#Searching for the most similar unit to "neural"
+head(rownames(c.SIM[order(c.SIM["neural",], decreasing = TRUE),]), n = 1)
 ```
 
```

Comment:
What do your components represent in this example and why are they more suitable for the recommender system than just using the variables?